engine/player/soulbinds.cpp

```diff
@@ -553,28 +572,36 @@ void combat_meditation( special_effect_t& effect )
   struct combat_meditation_buff_t : public stat_buff_t
   {
     timespan_t ext_dur;
-    combat_meditation_buff_t( player_t* p, double duration_mod, bool icd_enabled ) : stat_buff_t( p, "combat_meditation", p->find_spell( 328908 ) )
+    int triggered_times;
+
+    combat_meditation_buff_t( player_t* p, double duration_mod, bool icd_enabled )
+      : stat_buff_t( p, "combat_meditation", p->find_spell( 328908 ) )
     {
       set_cooldown( icd_enabled ? data().effectN( 3 ).trigger()->duration() : 0_ms );
       set_refresh_behavior( buff_refresh_behavior::EXTEND );
       set_duration_multiplier( duration_mod );
 
       ext_dur = duration_mod * timespan_t::from_seconds( p->find_spell( 328913 )->effectN( 2 ).base_value() );
 
+      triggered_times = 0;
+
       // TODO: add more faithful simulation of delay/reaction needed from player to walk into the sorrowful memories
       set_tick_callback( [ this ]( buff_t*, int, timespan_t ) {
-        if ( rng().roll( sim->shadowlands_opts.combat_meditation_extend_chance ) )
+        if ( this->triggered_times++ < 3 && rng().roll( sim->shadowlands_opts.combat_meditation_extend_chance ) )
           extend_duration( player, ext_dur );
       } );
```

Comment:
You'd want to increment the counter only if the roll for the chance was successful, unless you only ever get a maximum of 3 orbs to spawn?

Reply:
You only ever get a maximum of 3 orbs spawned